cmd/kube-scheduler/app/options/insecure_serving.go

```
@@ -0,0 +1,164 @@
+		return nil
+	}
+
+	if o.Healthz != nil {
+		o.Healthz.BindPort = o.BindPort
+		o.Healthz.BindAddress = net.ParseIP(o.BindAddress)
+	}
+	if o.Metrics != nil {
+		o.Metrics.BindPort = o.BindPort
+		o.Metrics.BindAddress = net.ParseIP(o.BindAddress)
+	}
+
+	return o.applyTo(c, componentConfig)
+}
+
+// ApplyToFromLoadedConfig updates the insecure serving options from the component config and then appies it to the given scheduler app configuration.
+func (o *CombinedInsecureServingOptions) ApplyToFromLoadedConfig(c *schedulerappconfig.Config, componentConfig *componentconfig.KubeSchedulerConfiguration) error {
+	if o == nil {
+		return nil
+	}
+
+	if err := updateInsecureServingOptionsFromAddress(o.Healthz, componentConfig.HealthzBindAddress); err != nil {
+		return fmt.Errorf("invalid healthz address: %v", err)
+	}
+	if err := updateInsecureServingOptionsFromAddress(o.Metrics, componentConfig.MetricsBindAddress); err != nil {
+		return fmt.Errorf("invalid metrics address: %v", err)
+	}
+
+	return o.applyTo(c, componentConfig)
+}
+
+func updateAddressFromInsecureServingOptions(addr *string, is *controlleroptions.InsecureServingOptions) error {
+	if is == nil {
+		*addr = ""
+	} else {
+		if is.Listener != nil {
+			*addr = is.Listener.Addr().String()
+		} else if is.BindPort == 0 {
+			*addr = ""
+		} else {
+			*addr = net.JoinHostPort(is.BindAddress.String(), strconv.Itoa(is.BindPort))
+		}
+	}
+
+	return nil
+}
+
+func updateInsecureServingOptionsFromAddress(is *controlleroptions.InsecureServingOptions, addr string) error {
+	if is == nil {
+		return nil
+	}
+	if len(addr) == 0 {
+		is.BindPort = 0
+		return nil
+	}
+
+	host, portInt, err := splitHostIntPort(addr)
+	if err != nil {
+		return fmt.Errorf("invalid address %q", addr)
+	}
+
+	is.BindAddress = net.ParseIP(host)
+	is.BindPort = portInt
+
+	return nil
+}
+
+// Validate validates the insecure serving options.
+func (o *CombinedInsecureServingOptions) Validate() []error {
+	if o == nil {
+		return nil
+	}
```

for this pattern - would it make more sense to return an error in the case where the object is `nil`, or even let it panic unless calls on `nil` objects actually happen in the call stack?

---

This is intentional, a pattern we follow everywhere else: if you want to disable a feature completely, set the options to nil.